File: Circles/VibeCoding-U30C15L1/event-planner/frontend/src/types/event.ts
export interface BaseEvent {
  id: string;
  title: string;
  description: string;
  location: string;
  start_time: string;
  end_time: string;
  is_public: boolean;
  category: string;
  max_attendees?: number;
  created_by: string;
  created_at: string;
  updated_at: string;
  image_url?: string;
}

export interface Event extends BaseEvent {
  creator: User | null;
  attendees_count: number;
  current_user_rsvp: RSVP | null;
}

export interface User {
  id: string;
  email: string;
  full_name: string;
  avatar_url?: string;
}


export interface RSVP {
  id: string;
  event_id: string;
  user_id: string;
  status: 'going' | 'not_going' | 'maybe';
  created_at: string;
  updated_at: string;
}

File: Circles/VibeCoding-U30C15L1/event-planner/backend/app/models/event.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional, List
from .user import UserResponse

class EventBase(BaseModel):
    title: str = Field(..., min_length=3, max_length=100)
    description: Optional[str] = None
    location: Optional[str] = None
    start_time: datetime
    end_time: datetime
    is_public: bool = True
    category: Optional[str] = None
    max_attendees: Optional[int] = None

class EventCreate(EventBase):
    pass

class EventUpdate(EventBase):
    title: Optional[str] = None
    start_time: Optional[datetime] = None
    end_time: Optional[datetime] = None

class EventInDB(EventBase):
    id: str
    created_by: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

class EventResponse(EventInDB):
    creator: Optional[UserResponse] = None
    attendees_count: int = 0
    current_user_rsvp: Optional[str] = None
